src/seat.c

```diff
@@ -352,26 +352,40 @@ focus_change_notify(struct wl_listener *listener, void *data)
 	struct view *view = event->new_surface ?
 		view_from_wlr_surface(event->new_surface) : NULL;
 
+	/* Prevent focus switch to layershell client from updating view state */
+	if (!view && server->active_view && event->new_surface
+			&& wlr_layer_surface_v1_try_from_wlr_surface(event->new_surface)) {
+		return;
+	}
+
+	/*
+	 * Due to the previous conditions we may end up in a situation where
+	 * we give focus to a previously focused view that was never deactivated
+	 */
+	if (view == server->active_view) {
```

Comment:
This early-out (`view == server->active_view`) is redundant with the reverse condition (`view != server->active_view`) later on. Let's remove one or the other.

Reply:
Good catch. Removed the first one.